### PR TITLE
Issue #2428: removed dependency from commons-lang3

### PR DESCRIPTION
**Pull request by chrismacias**
Added necessary methods from ArrayUtils in commons-lang3 to CommonUtils class and changed dependencies throughout the project where needed

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 1]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final Class<?>[] EMPTY_CLASS_ARRAY = new Class[0];
+    /**
+     * An empty immutable {@code String} array.
+     */
+    public static final String[] EMPTY_STRING_ARRAY = new String[0];
+    /**
+     * An empty immutable {@code long} array.
+     */
+    public static final long[] EMPTY_LONG_ARRAY = new long[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l407)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 2]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final byte[] EMPTY_BYTE_ARRAY = new byte[0];
+    /**
+     * An empty immutable {@code Byte} array.
+     */
+    public static final Byte[] EMPTY_BYTE_OBJECT_ARRAY = new Byte[0];
+    /**
+     * An empty immutable {@code double} array.
+     */
+    public static final double[] EMPTY_DOUBLE_ARRAY = new double[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l439)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 3]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final float[] EMPTY_FLOAT_ARRAY = new float[0];
+    /**
+     * An empty immutable {@code Float} array.
+     */
+    public static final Float[] EMPTY_FLOAT_OBJECT_ARRAY = new Float[0];
+    /**
+     * An empty immutable {@code boolean} array.
+     */
+    public static final boolean[] EMPTY_BOOLEAN_ARRAY = new boolean[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l455)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 4]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final Byte[] EMPTY_BYTE_OBJECT_ARRAY = new Byte[0];
+    /**
+     * An empty immutable {@code double} array.
+     */
+    public static final double[] EMPTY_DOUBLE_ARRAY = new double[0];
+    /**
+     * An empty immutable {@code Double} array.
+     */
+    public static final Double[] EMPTY_DOUBLE_OBJECT_ARRAY = new Double[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l443)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 5]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final boolean[] EMPTY_BOOLEAN_ARRAY = new boolean[0];
+    /**
+     * An empty immutable {@code Boolean} array.
+     */
+    public static final Boolean[] EMPTY_BOOLEAN_OBJECT_ARRAY = new Boolean[0];
+    /**
+     * An empty immutable {@code char} array.
+     */
+    public static final char[] EMPTY_CHAR_ARRAY = new char[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l463)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 6]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final String[] EMPTY_STRING_ARRAY = new String[0];
+    /**
+     * An empty immutable {@code long} array.
+     */
+    public static final long[] EMPTY_LONG_ARRAY = new long[0];
+    /**
+     * An empty immutable {@code Long} array.
+     */
+    public static final Long[] EMPTY_LONG_OBJECT_ARRAY = new Long[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l411)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 7]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final short[] EMPTY_SHORT_ARRAY = new short[0];
+    /**
+     * An empty immutable {@code Short} array.
+     */
+    public static final Short[] EMPTY_SHORT_OBJECT_ARRAY = new Short[0];
+    /**
+     * An empty immutable {@code byte} array.
+     */
+    public static final byte[] EMPTY_BYTE_ARRAY = new byte[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l431)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 8]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final Integer[] EMPTY_INTEGER_OBJECT_ARRAY = new Integer[0];
+    /**
+     * An empty immutable {@code short} array.
+     */
+    public static final short[] EMPTY_SHORT_ARRAY = new short[0];
+    /**
+     * An empty immutable {@code Short} array.
+     */
+    public static final Short[] EMPTY_SHORT_OBJECT_ARRAY = new Short[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l427)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 9]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+    //from document CommonUtils.java
+    ///////////////////////////////////////////////////////////////////////////////////////
+    /**
+     * An empty immutable {@code Object} array.
+     */
+    public static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+    /**
+     * An empty immutable {@code Class} array.
+     */
+    public static final Class<?>[] EMPTY_CLASS_ARRAY = new Class[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l399)

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java [hunk 10]
```diff
@@ -384,4 +384,125 @@ public static String fillTemplateWithStringsByRegexp(
...
+     */
+    public static final Float[] EMPTY_FLOAT_OBJECT_ARRAY = new Float[0];
+    /**
+     * An empty immutable {@code boolean} array.
+     */
+    public static final boolean[] EMPTY_BOOLEAN_ARRAY = new boolean[0];
+    /**
+     * An empty immutable {@code Boolean} array.
+     */
+    public static final Boolean[] EMPTY_BOOLEAN_OBJECT_ARRAY = new Boolean[0];
```
Comment (mkordas) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/checkstyle/checkstyle/file/1192454837/issues/source?bid=2193633&fileBranchId=2924055#l459)

**Comment by romani** [comment]
@chrismacias , please remove from CommonUtils all constants that are not in use right now , example: "EMPTY_SHORT_OBJECT_ARRAY".
All failures on CIs have to be resovled, all comments from mkordas are optional and most likely duplicates of CIs.

**Pull request closed without merging**